src/main.rs

```diff
@@ -112,16 +112,16 @@ impl LineList {
     }
 
     fn handle_key(&mut self, key: i32, cmdline_result: &Result<String, pcre2::Error>,
-                  global: &mut Global) -> Result<(), Box<dyn Error>> {
+                  global: &mut Global, profile: &Profile) -> Result<(), Box<dyn Error>> {
         if !global.handle_key(key) {
             match key {
                 KEY_RETURN => {
                     if let Ok(cmdline) = cmdline_result {
                         // TODO(#47): endwin() on Enter in LineList looks like a total hack and it's unclear why it even works
                         endwin();
-                        // TODO(#40): shell is not customizable
                         // TODO(#50): cm doesn't say anything if the executed command has failed
-                        Command::new("sh")
+                        let shell = profile.shell.as_ref();
+                        Command::new(shell.unwrap_or(&"sh".to_string()))
                             .stdin(File::open("/dev/tty")?)
                             .arg("-c")
```

Comment:
What if the custom shell does not support this flag? How should we address that? Should we address that?

Reply:
We could accept whole shell expression in the config (i.e. `sh -c` or `cmd.exe /c`), split it by spaces, and use `.args` method on `Command` to pass any arguments.